### ISSUE TITLE
Problem playing YouTube videos

### ISSUE THREAD

**Issue by lfom** [tuyafeng/Via#596]
Since a few versions ago (I am not sure when exactly, unfortunately), YouTube videos don't play correctly on Via anymore. I can hear the sound but the video is either frozen, black or blinks a few frames. Via installed on Android 9, latest Android Webview installed. I tried clearing cache/cookies and it didn't help. By the way, thanks for this great little browser. Eagerly waiting for the Reader mode and force zoom option. Regards.

**Comment by tuyafeng** [comment]
Are your user agent settings the default? It is best to confirm the version of Via, the latest version is 4.0.5.

**Comment by lfom** [comment]
> Are your user agent settings the default? It is best to confirm the version of Via, the latest version is 4.0.5.

Sorry: yes, I am using latest version (4.0.5) but I also had the same problem in the previous version, and Via is using the default UA.

**Comment by Cutetlaem** [comment]
> > Cài đặt tác nhân người dùng của bạn có phải là cài đặt mặc định không? Tốt nhất là xác nhận phiên bản của Via, phiên bản mới nhất là 4.0.5.
> 
> Xin lỗi: có, tôi đang sử dụng phiên bản mới nhất (4.0.5) nhưng tôi cũng gặp sự cố tương tự ở phiên bản trước và Via đang sử dụng UA mặc định.

I am also using 2 devices: Android 9 and Android 10. The version I edited has blocked the class attribute "ad-showing" on YouTube, everything is working properly, no ads appear. However, the YouTube version of the phone browser has one limitation: When playing a video, sometimes you get a request to press play video to turn on the sound, which causes the ad blocker to crash. The problem is that the video playback screen is black and the video cannot be played. The solution is to press reload the page, everything will be normal.

**Comment by lfom** [comment]
I have latest Google WebView installed, and as I said previously I can hear the sound without problems so I guess the app or streaming didn't crash. It only happens with YouTube, other streaming video sites work, like BitChute or Libry.tv.

**Comment by Cutetlaem** [comment]
> I have latest Google WebView installed, and as I said previously I can hear the sound without problems so I guess the app or streaming didn't crash. It only happens with YouTube, other streaming video sites work, like BitChute or Libry.tv.
> Edit: I also tried to clear WebView's data (since nothing is important there anyway) and cache/cookies from Via, but it didn't help. Ad block is disabled in Via by the way, since I use another program to block ads and secure DNS, but disabling the app didn't change anything.

Android WebView is a system application, should you try to uninstall the update then update it again? Does the same thing happen when you use another browser application?
I don't have the source code to find out the cause, so wait for the update or wait for the project owner to respond.
Good luck.

**Comment by tuyafeng** [comment]
Sorry for the late reply. But I still can not find the reason. Can you tell me about your phone model? And that will be great if you can post a screenshot about youtube page.

**Comment by lfom** [comment]
> Sorry for the late reply. But I still can not find the reason. Can you tell me about your phone model? And that will be great if you can post a screenshot about youtube page.

![Screenshot_20200915_045534_mark.via.gp.jpg](https://user-images.githubusercontent.com/15659999/93182771-786a6f00-f710-11ea-8d89-9d729fa25bce.jpg). ![Screenshot_20200915_045636_com.android.settings.jpg](https://user-images.githubusercontent.com/15659999/93182801-81f3d700-f710-11ea-96d0-bed8333729d6.jpg)

**Comment by lfom** [comment]
Finally I have found a way to play YT videos on Via again: it seems to be a problem between Android Webview and Chrome Webview. I noticed that when playing YT videos on other browsers it would show "Chrome Mobile Webview" in the debug information, while Via would show "Android Webview" (I am not sure about the exact name, but it wasn't Chrome). On EMUI 9.1 I can see both but it is only possible to select Google Webview: selecting Chrome Webview would show a toast message and the setting would got back to Google. Then I went to phone settings and uninstalled Android System Updates (I guess this is listed as Google Webview in the other setting) and Via started to play YT videos normally again. If I update Android Webview then the problem starts again, even if the debug information shows that Via is using Chrome Mobile Webview. I guess I will not update Android Webview for now.
